static/usr/bin/core-sshd-host-keygen

```diff
@@ -63,4 +61,7 @@ create_keys() {
 		"$hostkeys" /etc/ssh/ssh_host_ed25519_key -t ed25519
 }
 
-create_keys
+# create keys only if we are in run mode, there is no use to create keys in install mode
+if [ -n "$(grep "snapd_recovery_mode=install" /proc/cmdline)" ]; then
+  create_keys
+fi
```

Comment:
what about recovery mode ?

Reply:
that is, should we also skip generation for recovery mode as well? Ideally, recover mode would have the same host keys as from run mode by copying the files from ubuntu-data over to the tpmfs from the initramfs like we do for other things

Reply:
I was having same dilemma. But I was not sure what system is there in recovery mode.
Ultimately I opt for small step, as install mode is clear case and generated key is thrown away.

Also install mode is something which occurs on every first boot and in the factory, so extra time there is costly.
In comparison, recovery mode is rare in comparison, and more "please can you get me out of the trouble, I don't really care if it takes extra seconds, just help  me"

But happy to include recovery mode, as ultimately we end up in run mode and will genera they key there. You tell me preferred one :)